pkg/platform/runtime/setup/implementations/alternative/runtime.go

```diff
@@ -25,7 +25,9 @@ func NewSetup(store *store.Store, artifacts artifact.ArtifactRecipeMap) *Setup {
 	return &Setup{store: store, artifacts: artifacts}
 }
 
-func (s *Setup) DeleteOutdatedArtifacts(changeset artifact.ArtifactChangeset, storedArtifacted store.StoredArtifactMap) error {
+func (s *Setup) DeleteOutdatedArtifacts(changeset artifact.ArtifactChangeset, storedArtifacted store.StoredArtifactMap) ([]artifact.ArtifactID, error) {
+	var alreadyInstalled []artifact.ArtifactID
```

Comment:
Not going to request changes on this now, but this seems like something we should fix in the future. It's awkward that a function responsible for deleting outdated artifacts is returning artifacts that you want to keep. Either this function needs to be renamed or its logic needs to be split up into separate functions.

Reply:
I agree.  I pushed a new version, that also gets rid of the go-funk dependency...